charts/rancher-vsphere-csi/rancher-vsphere-csi/2.1.1/Chart.yaml

```
@@ -1,19 +1,21 @@
 annotations:
+  catalog.cattle.io/rancher-version: <2.5.99-0
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: vSphere CSI
   catalog.cattle.io/namespace: kube-system
   catalog.cattle.io/os: linux
   catalog.cattle.io/release-name: vsphere-csi
 apiVersion: v1
-appVersion: 2.1.0
+appVersion: 2.1.1
 description: vSphere Cloud Storage Interface (CSI)
 icon: https://charts.rancher.io/assets/logos/vsphere-csi.svg
 keywords:
 - infrastructure
+kubeVersion: ==1.18
```

typo? Do you mean `>=`?

---

Nope, we want this chart to be available for 1.18 only. Since CPI and CSI are meant to be used together, and the new CPI supports 1.18 - 1.21, but new CSI 1.19 - 1.21. There is one particular case where you need the new CPI chart with the old CSI 2.1.0 to be fully compatible in k8s 1.18.

Pairings:
k8s <= 1.17: New installs will be unsupported, users will need to upgrade k8s to at least 1.18.
k8s == 1.18: CPI 1.0.1 + CSI 2.1.1
k8s >= 1.19: CPI 1.0.1 + CSI 2.3.0

---

Cool, that makes sense. 

I am not sure if  `==1.18` is the right format as it is not listed in Helm's docs: https://helm.sh/docs/topics/charts/#the-kubeversion-field
```suggestion
kubeVersion: = 1.18
```

---

Ah, there it is, I was looking for an example of that. The `==` does work since I ended up testing it to make sure and I get the same result with both. I'll change it to `=` thought, since it's what the docs say so I prefer it.